### PR TITLE
Master

### PR DESCRIPTION
**Pull request by anotheredward**
Update production with the latest version of the code.

I was looking to have an architecture similar to the Elm architecture: http://guide.elm-lang.org/architecture/index.html

Code is all meant to be ES5 compliant.

### CODE REVIEW

File: js/data.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      const lines = convertSignsToLines(signs)
+      const rows = R.pipe(
+        R.splitEvery(2),
+        R.map(R.zipObj(['name', 'time']))
+      )(lines)
+      callback(rows)
+    }
+  ).fail(function (error) {
+    console.log(error, new Date())
+    $('#loading').html(error.statusText)
```
Comment (anotheredward) [inline js/data.js]: The statusText is "error", pretty underwhelming :(.
The real error gets printed to console.error from inside jQuery and I can't catch it :(.

File: js/render.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+function render(rows) {
+  $('.travel-list').empty()
+  $('.travel-list').append('<tr>'
+    + '<th class="route_col">From AMA to</th>'
+    + '<th class="mins_col">Mins</th>'
+    + '<th class="trend_col"></th>'
+    + '</tr>')
+
+  const formattedRows = rows.map(function(row, key) {
```
Comment (matt-in-a-hat) [inline js/render.js]: Is `map` within your target js version?

File: js/data.js (hunk 1 above)
Comment (matt-in-a-hat) [inline js/data.js]: Do we want to display a nicer error, like 'Service unavailable' ?

File: js/data.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/* global $, R */
+var SIGN_IDS = [
+  369,
+  642
+]
+
+function getRows(callback) {
```
Comment (matt-in-a-hat) [inline js/data.js]: Are you just defining these on the global/window scope to call them from other files? Is there a more explicit way of doing this?

**Comment by matt-in-a-hat** [comment]
So at this stage there isn't any configuration? This is just a prototype for 1 board? Are you thinking the ids and mall name would come from URL params?

File: js/render.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  + '<td class="route_col">' + name + '</td>'
+  + '<td class="mins_col">' + newTime + '</td>'
+  + '<td class="trend_col">' + getChevron(currentTime, newTime) + '</td>'
+  + '</tr>'
+}
+
+function getChevron (currentTime, newTime) {
+  if (currentTime < newTime) return '<i class="fa fa-chevron-up"></i>'
+  if (currentTime > newTime) return '<i class="fa fa-chevron-down"></i>'
+  return ''
```
Comment (matt-in-a-hat) [inline js/render.js]: No keeping history for trend yet, just comparing to 30 seconds ago?

File: js/data.js (hunk 1 above)
Comment (anotheredward) [inline js/data.js]: Will do :)

File: js/render.js (hunk 2 above)
Comment (anotheredward) [inline js/render.js]: Map is ES5

File: js/data.js (hunk 3 above)
Comment (anotheredward) [inline js/data.js]: Could wrap them in an object or the module pattern, but I see little benefit for now.

File: js/render.js (hunk 4 above)
Comment (anotheredward) [inline js/render.js]: Correct, there is no trend, and no configuration for the time being :)

**Comment by matt-in-a-hat** [comment]
Looks like a good start. Haven't tested, assume the xpath and unreadable ramda voodoo do what you're expecting.

**Comment by anotheredward** [comment]
lol @ unreadable ramda voodoo.

**Pull request merged**